server/queue/queue-rabbitmq/src/main/java/org/apache/james/queue/rabbitmq/view/cassandra/CassandraMailQueueView.java

```diff
@@ -39,10 +41,16 @@
         private final CassandraMailQueueMailDelete cassandraMailQueueMailDelete;
 
         @Inject
-        public Factory(CassandraMailQueueMailStore storeHelper, CassandraMailQueueBrowser cassandraMailQueueBrowser, CassandraMailQueueMailDelete cassandraMailQueueMailDelete) {
+        public Factory(CassandraMailQueueMailStore storeHelper,
+                       CassandraMailQueueBrowser cassandraMailQueueBrowser,
+                       CassandraMailQueueMailDelete cassandraMailQueueMailDelete,
+                       EventsourcingConfigurationManagement eventsourcingConfigurationManagement,
+                       CassandraMailQueueViewConfiguration configuration) {
             this.storeHelper = storeHelper;
             this.cassandraMailQueueBrowser = cassandraMailQueueBrowser;
             this.cassandraMailQueueMailDelete = cassandraMailQueueMailDelete;
+
+            eventsourcingConfigurationManagement.loadConfiguration(configuration);
```

Comment:
once loadConfiguration will be properly named (to register for example), it won't make sense anymore that the MQFactory register the configuration IMO.

Reply:
Do we agree the MQ system should not start with an invalid configuration?

Reply:
> Do we agree the MQ system should not start with an invalid configuration?

Yes, and it's interesting because in your wording, an invalid configuration could be a "valid one incompatible with the previous one".
It would mean that a configuration is not only a single configuration  Pojo but 2 : the current one and the (optional) previous one.

Could we make a design choice about what a configuration is ?

The code will obviously follow the design ...